lib/buffer.js

```diff
@@ -107,7 +111,7 @@ function Buffer(arg, encodingOrOffset, length) {
         'If encoding is specified then the first argument must be a string'
       );
     }
-    return Buffer.allocUnsafe(arg);
+    return allocate(arg);
```

Comment:
@RReverser #7051 introduced that check for negative numbers into `allocate()`, so it should be fine (which is one of the reasons I think it’s okay to keep `allocate()` below in `.concat()). So yes, I agree that that’s a good idea. ;)